### ISSUE TITLE
Cannot add Anki cards on Brave

### ISSUE THREAD

**Issue by VoQZHD** [themoeway/yomitan#335]
![image](https://github.com/themoeway/yomitan/assets/45748775/3df7262d-9588-461c-8f69-5ed199d77445)
![image](https://github.com/themoeway/yomitan/assets/45748775/cc8a2cab-3061-4b3f-bf79-d5e21b2fe99f)
![image](https://github.com/themoeway/yomitan/assets/45748775/a63efe14-d457-4915-89d4-64724ea5fbd3)

The buttons are green but appear to do nothing when pressed, there's a progress indicator but no progress.

Brave Version 1.60.118 Chromium: 119.0.6045.163 (Official Build)  (64-bit)

Yomitan (development build) 23.11.23.0
[yomitan-settings-2023-11-29-15-22-38.json](https://github.com/themoeway/yomitan/files/13502312/yomitan-settings-2023-11-29-15-22-38.json)

**Comment by Cyanisk** [comment]
I seem to have the exact same problem with brave.
Yomitan is connected to Anki, can find duplicates and open the Anki browser to display said duplicates.
However the progress bar keeps spinning and (after about 5 minutes) finally times out:

![2023-12-15_15-12](https://github.com/themoeway/yomitan/assets/101006560/73029268-6e2e-4293-aef4-a132af0c7e31)

`An error occurred:
Unexpected response of type undefined ({"action":"injectAnkiNoteMedia","params":{"timestamp":1702634261405,"definitionDetails":{"type":"term","term":"食べる","reading":"たべる"},"audioDetails":{"sources":[{"type":"jpod101","url":"","voice":""}],"preferredAudioIndex":null,"idleTimeout":null},"screenshotDetails":null,"clipboardDetails":{"image":false,"text":false},"dictionaryMediaDetails":[]}})
Log debug info to console`

Clicking `Log debug info to console` logs a `{ankiNoteErrors: Array(1)}` to the developer tools console, but I don't know if it's helpful (and if so, how to export it)?

For good measure I tried using Yomitan in ungoogled-chromium which seems to work without issue

Brave Version 1.61.101 Chromium: 120.0.6099.71 (Official Build) (64-bit)
Yomitan 23.11.23.0

**Comment by bijak** [comment]
Same problem here, also on Brave
Version 1.61.104 Chromium: 120.0.6099.115 (Official Build)
Yomitan: 23.11.23.0

**Comment by wolfv506** [comment]
Me too, I have the same specs as bijak

**Comment by djahandarie** [comment]
Sounds like the problem is limited to Brave. I guess we don't have any developers who use it... we'd need to find someone willing to debug.

**Comment by bijak** [comment]
For the record, reinstalling the extension did not resolve this issue for me

**Comment by MangoHusky** [comment]
Reinstalling the extension also didn't solve resolve it for me. I even switched from stable to testing, but it still doesn't work.

**Comment by tahoeschrader** [comment]
I am not sure why this issue was closed. I have the same issue, and reinstalling the extension does not fix it.

**Comment by djahandarie** [comment]
If anyone viewing this issue has Brave and time to debug, please try to makes some progress and leave notes about what the issue might be 🙏

**Comment by tahoeschrader** [comment]
I don't have any Javascript or Chrome Dev Tools experience, but the card creation starts to hang at line 63 of `js/comm/api.js`

```javascript
return this._invoke('injectAnkiNoteMedia', {timestamp, definitionDetails, audioDetails, screenshotDetails, clipboardDetails, dictionaryMediaDetails});
```

![Screenshot 2023-12-25 at 6 26 31 PM](https://github.com/themoeway/yomitan/assets/26147461/d86464e9-3175-4dc1-ac65-d28c66dc9124)

This is basically just saying exactly what's written out in the error from @Cyanisk up above. If anyone has advice on where I could like, I can keep trying. Once it hits `chrome.runtime.sendMessage(...)`, I am unable to step into any more functions and the process starts taking forever to load.

**Comment by tahoeschrader** [comment]
When I try debugging on Chrome Canary, it gets past this step with no issue even though all the Watch objects look the same. I also couldn't debug into `chrome.runtime.sendMessage(...)`, so wasn't able to get any helpful information. Like others mentioned, there is no issue with Yomibaba on Brave.

**Comment by djahandarie** [comment]
Thanks for trying to debug! That's just the message that gets sent between the yomitan content script and yomitan service worker. If you open up your extensions, there should be a way to inspect the service worker page. On that page, put a breakpoint in backend.js on the appropriate function (should be called _onApiInjectAnkiNoteMedia IIRC) -- that should let you continue debugging until the service worker hits some problem.

**Comment by Cyanisk** [comment]
Yomitan has been working flawlessly for me after simply reinstalling the extension, so unfortunately I don't think I'll be able to help debugging

**Comment by tahoeschrader** [comment]
Thanks djahandarie, I was able to get further with that advice. Tracking down the problem right now and it seems to be coming from `_downloadAudioFromUrl`

**Comment by bijak** [comment]
Wasn't able to get further than above tracing error messages, but I did figure out some steps to reproduce the problem.
1. Set up Yomitan and Anki so that everything is working
2. Restart your computer *or* fully restart Brave (ie kill it in the task manager after closing the browser window since it keeps running in the background)
3. Now you can no longer mine to Anki
I tried this several times and it broke it every time. Additionally I found a much simpler fix than reinstalling the extension: just disabling and re-enabling it again in the extension options.

I _also_ tried reproducing these steps on Edge since I have it installed, and no luck there so it does seem to be something Brave specific.

**Comment by wolfv506** [comment]
Like discussed on the discord, the problem only arises if you have an {audio} field in your anki export template. (maybe helpful for people who want to reproduce the bug)

**Comment by AccountForStealingCode** [comment]
try modifying this option
![image](https://github.com/themoeway/yomitan/assets/119428043/eb1fc119-420c-4ab7-94ba-1d14f57696fa)
also applies to firefox

**Comment by tahoeschrader** [comment]
> try modifying this option
> 
> ![image](https://github.com/themoeway/yomitan/assets/119428043/eb1fc119-420c-4ab7-94ba-1d14f57696fa)
> 
> also applies to firefox

changing this setting to all the available options doesn't seem to affect this bug

**Comment by AccountForStealingCode** [comment]
> changing this setting to all the available options doesn't seem to affect this bug
have you tried turning the extension off and on again

**Comment by tahoeschrader** [comment]
> > changing this setting to all the available options doesn't seem to affect this bug
> 
> have you tried turning the extension off and on again

yes (others in this issue have also tried reinstalling to no avail)

![image](https://github.com/themoeway/yomitan/assets/26147461/de6b16c3-ffae-4c58-ab38-673133e8fb55)

**Comment by AccountForStealingCode** [comment]
> yes (others in this issue have also tried reinstalling to no avail)

try turning off the actual extension
![image](https://github.com/themoeway/yomitan/assets/119428043/ed3fe854-6270-4535-a720-745e7ac01e42)

https://github.com/themoeway/yomitan/assets/119428043/31f6ac28-34bc-4be7-af11-32421d16011e

**Comment by wolfv506** [comment]
> > yes (others in this issue have also tried reinstalling to no avail)
> 
> try turning off the actual extension ![image](https://private-user-images.githubusercontent.com/119428043/293571186-ed3fe854-6270-4535-a720-745e7ac01e42.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQxMDYzMTcsIm5iZiI6MTcwNDEwNjAxNywicGF0aCI6Ii8xMTk0MjgwNDMvMjkzNTcxMTg2LWVkM2ZlODU0LTYyNzAtNDUzNS1hNzIwLTc0NWU3YWMwMWU0Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwMVQxMDQ2NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yOGNiN2U2ZjdjMjIyYzA5ZGQ3ZGIxZTY3MzYxN2VjNjA1ZDNlODVmMzRhOGQ0YTA0OTlhNjI4ZDRmZjI2YTk4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.-5Zfwaip7C9UoRoR1k1qARR9Xt8lmjP08Ic82wlO-yU)
> 
>  2024-01-01.13-25-22.mp4

This works but when I restart my PC, I have the same bug again...

**Comment by scarrupt** [comment]
It might due to the following bug with declarativeNetRequest:

https://github.com/brave/brave-browser/issues/30854

Fix should be added in Brave version 1.63

**Comment by arbyste** [comment]
I can confirm that I have this issue in Brave 1.61, but not in 1.63 (beta version)

**Comment by Katsumi-III** [comment]
Also have this issue and attempted to install the recommended version of Brave. but yielded
```
 Could not depsolve transaction; 1 problem detected:
 Problem: conflicting requests
  - nothing provides brave-keyring >= 1.11-1 needed by brave-browser-beta-1.63.134-1.x86_64 from @commandline
```

As a layman just trying to use ASBPLAYER with Yomitan here, I don't know what I am looking at, nor how to fix this

**Comment by Katsumi-III** [comment]
> Also have this issue and attempted to install the recommended version of Brave. but yielded
> 
> ```
>  Could not depsolve transaction; 1 problem detected:
>  Problem: conflicting requests
>   - nothing provides brave-keyring >= 1.11-1 needed by brave-browser-beta-1.63.134-1.x86_64 from @commandline
> ```
> 
> As a layman just trying to use ASBPLAYER with Yomitan here, I don't know what I am looking at, nor how to fix this

I had reinstalled Yomitan while on a previous version of Brave which yielded no resolution, but I have just reinstalled on Brave v1.62.153 and now Yomitan can successfully add new cards

**Comment by Katsumi-III** [comment]
Thought that it was resolved after reinstalling in Brave v1.62.153 but, after pc restart the issue reappeared.

**Comment by Katsumi-III** [comment]
> Thought that it was resolved after reinstalling in Brave v1.62.153 but, after pc restart the issue reappeared.

Re-enabling the extension seems to fix it temporarily

**Comment by jamesmaa** [comment]
Stale issue. Please reopen if you experience this problem
